x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts

```diff
@@ -0,0 +1,972 @@
+      environment: normalizeValue<string>(fields[SERVICE_ENVIRONMENT]),
+    },
+    processor: {
+      event: normalizeValue<string>(fields[PROCESSOR_EVENT]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+    },
+  };
+};
+
+// todo: pending #192337
+export const linkedParentsOfSpanMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  return {
+    span: {
+      links: [
+        {
+          trace: {
+            id: normalizeValue<string>(fields[SPAN_LINKS_TRACE_ID]),
+          },
+          span: {
+            id: normalizeValue<string>(fields[SPAN_LINKS_SPAN_ID]),
+          },
+        },
+      ],
+    },
+  };
+};
+
+export const transactionMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  return {
+    transaction: {
+      representative_count: normalizeValue<number>(fields[TRANSACTION_REPRESENTATIVE_COUNT]),
+      result: normalizeValue<string>(fields[TRANSACTION_RESULT]),
+      sampled: normalizeValue<boolean>(fields[TRANSACTION_SAMPLED]),
+      id: normalizeValue<string>(fields[TRANSACTION_ID]),
+      duration: {
+        us: normalizeValue<number>(fields[TRANSACTION_DURATION]),
+      },
+      type: normalizeValue<string>(fields[TRANSACTION_TYPE]),
+      name: normalizeValue<string>(fields[TRANSACTION_NAME]),
+    },
+    service: {
+      node: {
+        name: normalizeValue<string>(fields[SERVICE_NODE_NAME]),
+      },
+      language: {
+        name: normalizeValue<string>(fields[SERVICE_NODE_NAME]),
+      },
+      name: normalizeValue<string>(fields[SERVICE_NAME]),
+      framework: {
+        name: normalizeValue<string>(fields[SERVICE_FRAMEWORK_NAME]),
+      },
+    },
+    trace: {
+      id: normalizeValue<string>(fields[TRACE_ID]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+      version: normalizeValue<string>(fields[AGENT_VERSION]),
+    },
+    event: {
+      success_count: normalizeValue<boolean>(fields[EVENT_SUCCESS_COUNT]),
+      outcome: normalizeValue<EventOutcome>(fields[EVENT_OUTCOME]),
+    },
+    processor: {
+      event: normalizeValue<string>(fields[PROCESSOR_EVENT]),
+    },
+    data_stream: {
+      namespace: normalizeValue<string>(fields[DATA_STREAM_NAMESPACE]),
+      type: normalizeValue<string>(fields[DATA_STEAM_TYPE]),
+      dataset: normalizeValue<string>(fields[DATA_STREAM_DATASET]),
+    },
+    span: {
+      id: normalizeValue<string>(fields[SPAN_ID]),
+    },
+    observer: {
+      hostname: normalizeValue<string>(fields[OBSERVER_HOSTNAME]),
+      type: normalizeValue<string>(fields[OBSERVER_TYPE]),
+      version: normalizeValue<string>(fields[OBSERVER_VERSION]),
+    },
+    timestamp: {
+      us: normalizeValue<number>(fields[TIMESTAMP]),
+    },
+    '@timestamp': normalizeValue<string>(fields[AT_TIMESTAMP]),
+    labels: {
+      some_resource_attribute: normalizeValue<string>(fields[LABEL_SOME_RESOURCE_ATTRIBUTE]),
+    },
```

Comment:
Similar to the  comment about HTTP headers, we should not hard-code specific labels and instead map any fields matching `labels.*`.